verifier/verifier.go

```diff
@@ -120,68 +130,89 @@ func NewVerifier(ociTrustPolicy *trustpolicy.OCIDocument, blobTrustPolicy *trust
-		if err != nil {
-			return nil, err
-		}
-	}
-
-	revocationTimestampClient := verifierOptions.RevocationTimestampClient
-	if revocationTimestampClient == nil {
-		var err error
-		revocationTimestampClient, err = revocation.NewTimestamp(&http.Client{Timeout: 2 * time.Second})
+	revocationTimestampingValidator := verifierOptions.RevocationTimestampingValidator
+	var err error
+	if revocationTimestampingValidator == nil {
+		revocationTimestampingValidator, err = revocation.NewWithOptions(revocation.Options{
+			OCSPHTTPClient:   &http.Client{Timeout: 2 * time.Second},
+			CertChainPurpose: x509.ExtKeyUsageTimeStamping,
+		})
 		if err != nil {
 			return nil, err
 		}
 	}
-
 	if trustStore == nil {
 		return nil, errors.New("trustStore cannot be nil")
 	}
-
 	if ociTrustPolicy == nil && blobTrustPolicy == nil {
 		return nil, errors.New("ociTrustPolicy and blobTrustPolicy both cannot be nil")
 	}
-
 	if ociTrustPolicy != nil {
 		if err := ociTrustPolicy.Validate(); err != nil {
 			return nil, err
 		}
 	}
-
 	if blobTrustPolicy != nil {
 		if err := blobTrustPolicy.Validate(); err != nil {
 			return nil, err
 		}
 	}
+	v := &verifier{
+		ociTrustPolicyDoc:               ociTrustPolicy,
+		blobTrustPolicyDoc:              blobTrustPolicy,
+		trustStore:                      trustStore,
+		pluginManager:                   pluginManager,
+		revocationTimestampingValidator: revocationTimestampingValidator,
+	}
 
-	return &verifier{
-		ociTrustPolicyDoc:         ociTrustPolicy,
-		blobTrustPolicyDoc:        blobTrustPolicy,
-		trustStore:                trustStore,
-		pluginManager:             pluginManager,
-		revocationClient:          revocationClient,
-		revocationTimestampClient: revocationTimestampClient,
-	}, nil
+	if err := v.setCodeSigningRevocation(verifierOptions); err != nil {
+		return nil, err
+	}
+	return v, nil
 }
 
-// NewFromConfig returns a OCI verifier based on local file system
+// NewFromConfig returns a OCI verifier based on local file system.
+//
 // Deprecated: NewFromConfig function exists for historical compatibility and should not be used.
 // To create an OCI verifier, use NewOCIVerifierFromConfig function.
 func NewFromConfig() (notation.Verifier, error) {
 	return NewOCIVerifierFromConfig()
 }
 
-// New creates a new verifier given ociTrustPolicy, trustStore and pluginManager
+// New creates a new verifier given ociTrustPolicy, trustStore and pluginManager.
+//
 // Deprecated: New function exists for historical compatibility and should not be used.
 // To create verifier, use NewVerifier function.
 func New(ociTrustPolicy *trustpolicy.OCIDocument, trustStore truststore.X509TrustStore, pluginManager plugin.Manager) (notation.Verifier, error) {
 	return NewVerifier(ociTrustPolicy, nil, trustStore, pluginManager)
 }
 
+// setCodeSigningRevocation sets code signing revocation object of v
+func (v *verifier) setCodeSigningRevocation(verifierOptions VerifierOptions) error {
```

Comment:
```suggestion
func (v *verifier) setRevocation(verifierOptions VerifierOptions) error {
```

Is it better to move `revocationTimestampingValidator ` to this function, so all revocation client related things are in one place.

Reply:
Updated.